src/dune/context.ml

```diff
@@ -35,6 +35,41 @@ module Env_nodes = struct
       (Dune_env.Stanza.find env_nodes.workspace ~profile).env_vars
 end
 
+let opam = Memo.Lazy.of_val (Bin.which ~path:(Env.path Env.initial) "opam")
+
+module Opam_config_var = struct
+  module Name = struct
+    include String
+
+    type t = string
+  end
+
+  module Value = struct
+    type t = string option
+
+    let to_dyn t = Dyn.Encoder.option String.to_dyn t
+
+    let equal x y = Option.equal String.equal x y
+  end
+
+  let read ~(kind : Kind.t) ~env (var : Name.t) : Value.t Fiber.t =
+    match var with
```

Comment:
Does it make sense to perhaps introduce a dependency on the current run? I'm not really sure if we can ever assume that an opam variable does not change between runs.

Reply:
Good point, I think we should indeed depend on the current run.